src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -44,6 +44,14 @@ public function load($resource, $type = null)
      */
     public function supports($resource, $type = null)
     {
-        return is_string($resource) && 'php' === pathinfo($resource, PATHINFO_EXTENSION);
+        if (!is_string($resource)) {
+            return false;
+        }
+
+        if (null === $type && 'php' === pathinfo($resource, PATHINFO_EXTENSION)) {
+            return true;
+        }
+
+        return null !== $type && 'php' === $type;
```

Comment:
``null !== $type`` should be removed in all loaders. It is useless as the condition ``'php' === $type`` already ensures is is not null